reactor-client/src/main/java/se/fortnox/reactivewizard/reactorclient/ReactorHttpClient.java

```diff
@@ -0,0 +1,735 @@
+        ReactorRxClientProvider clientProvider,
+        ObjectMapper objectMapper,
+        RequestParameterSerializers requestParameterSerializers,
+        Set<PreRequestHook> preRequestHooks
+    ) {
+        this.config = config;
+        this.clientProvider = clientProvider;
+        this.objectMapper = objectMapper;
+        this.objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        this.requestParameterSerializers = requestParameterSerializers;
+
+        serverInfo = new InetSocketAddress(config.getHost(), config.getPort());
+        collector = new ByteBufferCollector(config.getMaxResponseSize());
+        this.preRequestHooks = preRequestHooks;
+    }
+
+    public ReactorHttpClient(HttpClientConfig config) {
+        this(config, new ReactorRxClientProvider(config, new HealthRecorder()), new ObjectMapper(), new RequestParameterSerializers(), emptySet());
+    }
+
+    public static Mono<String> get(String url) {
+        try {
+            URL urlObj = new URL(url);
+
+            return reactor.netty.http.client.HttpClient.create()
+                .get()
+                .uri(urlObj.toString())
+                .responseContent()
+                .aggregate()
+                .asString();
+
+        } catch (MalformedURLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static void setTimeout(Object proxy, int timeout, TimeUnit timeoutUnit) {
+        if (Proxy.isProxyClass(proxy.getClass())) {
+            Object handler = Proxy.getInvocationHandler(proxy);
+            if (handler instanceof ReactorHttpClient) {
+                ((ReactorHttpClient)handler).setTimeout(timeout, timeoutUnit);
+            }
+        }
+    }
+
+    public void setTimeout(int timeout, TimeUnit timeoutUnit) {
+        this.timeout = timeout;
+        this.timeoutUnit = timeoutUnit;
+    }
+
+    public static <T> T create(HttpClientConfig config, ReactorRxClientProvider clientProvider, ObjectMapper objectMapper, Class<T> jaxRsInterface) {
+        return new ReactorHttpClient(config, clientProvider, objectMapper, new RequestParameterSerializers(), null).create(jaxRsInterface);
+    }
+
+    @SuppressWarnings("unchecked")
+    public <T> T create(Class<T> jaxRsInterface) {
+        return (T)Proxy.newProxyInstance(jaxRsInterface.getClassLoader(), new Class[]{jaxRsInterface}, this);
+    }
+
+    @Override
+    public Object invoke(Object proxy, Method method, Object[] arguments) {
+        if (arguments == null) {
+            arguments = new Object[0];
+        }
+
+        ReactorRequestBuilder request = createRequest(method, arguments);
+
+        addDevOverrides(request);
+        addAuthenticationHeaders(request);
+
+        reactor.netty.http.client.HttpClient rxClient = clientProvider.clientFor(request.getServerInfo());
+
+        reactor.netty.http.client.HttpClient.ResponseReceiver<?> response = ReactorHttpClient.submit(rxClient, request);
```

Comment:
ByteBuf instead of `?` perhaps? Investigate if we can avoid using `?` here

Reply:
Wrapped the content in a wrapper object so no questionmark types anymore for this one.